src/main/java/org/spongepowered/api/entity/ai/Goal.java

```diff
@@ -64,15 +64,15 @@
      * @param task The task to run
      * @return This goal, for chaining
      */
-    Goal<O> addTask(int priority, AITask<? extends O> task);
+    Goal<O> addTask(int priority, AITask<? super O> task);
```

Comment:
So you are passing in an AITask of unknown type, that may contain an O, (or another type) and turning that into a Goal<O>

That sounds like a casting error, an AITask of AITask<Object> might not even contain an O then you attempt to return a Goal<O> ?

Passing in an AITask<? extends O> means that you are guaranteed to have some sort of O inside AITask, you just don't know what the type is. Assuming that addTask doesn't need to know the type, returning Goal<O> is correct.